ta/mk/ta_dev_kit.mk

```diff
@@ -19,7 +19,8 @@ endif
 link-out-dir := $(out-dir)	# backward compat
 link-out-dir$(sm) := $(out-dir)
 
-binary := $(BINARY)
+user-ta-uuid := $(BINARY)
+user-ta-ldadd := $(LDADD)
```

Comment:
Like `LDFLAGS` that you have removed in a previous commit, I think `LDADD` is not the proper name to be used here because it does not deal with cross-compilation properly. It has happened in the past that some distribution set some compiler or linker variables by default (CFLAGS, LDFLAGS...), for use when building host binaries, which then conflicts with our cross-build.

How about using `LDADD_$(sm)` instead? That would be `LDADD_ta_arm32` or `LDADD_ta_arm64`, similar to what we already do with `CFLAGS_` and `CROSS_COMPILE_`.

Reply:
I was trying for a grace period for `BINARY` and `LDADD`. `LDADD` is used by at least on TA in optee_test.
I propose that we try to keep existing behavior with regards to `ta/mk/ta_dev_kit.mk` as much as possible in the PR.

Long term I'd like to replace what's in a TAs Makefile with what's in the TAs `user_ta.mk`, when we do that `LDADD` can be buried properly.

Reply:
OK